pkg/utils/logger/logger_test.go

```diff
@@ -0,0 +1,121 @@
+package logger_test
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/utils/logger"
+)
+
+type testWriter struct {
+	lastLog map[string]interface{}
+	levels  map[zerolog.Level]uint16
+}
+
+func (w *testWriter) WriteLevel(level zerolog.Level, p []byte) (n int, err error) {
+	if w.levels == nil {
+		w.levels = make(map[zerolog.Level]uint16)
+	}
+	w.levels[level]++
+	return w.Write(p)
+}
+
+func (w *testWriter) Write(p []byte) (n int, err error) {
+	if err = json.Unmarshal(p, &w.lastLog); err != nil {
+		return 0, err
+	}
+	return len(p), nil
+}
+
+func TestSeverityHook(t *testing.T) {
+	// Initialize zerolog with GCP logging requirements
+	zerolog.TimeFieldFormat = time.RFC3339
+	zerolog.TimestampFieldName = logger.GCPFieldKeyTime
+	zerolog.MessageFieldName = logger.GCPFieldKeyMsg
+
+	// Test writer
+	tw := &testWriter{}
+
+	// Add the severity hook for GCP logging
+	var gcpHook logger.SeverityHook
+	log.Logger = zerolog.New(tw).Hook(gcpHook).With().Timestamp().Logger()
+
+	log.Trace().Msg("just a trace")
+	require.Equal(t, uint16(1), tw.levels[zerolog.TraceLevel])
+	require.Contains(t, tw.lastLog, logger.GCPFieldKeySeverity)
+	require.Equal(t, "DEBUG", tw.lastLog[logger.GCPFieldKeySeverity])
+	require.Contains(t, tw.lastLog, logger.GCPFieldKeyMsg)
+	require.Equal(t, "just a trace", tw.lastLog[logger.GCPFieldKeyMsg])
+	require.Contains(t, tw.lastLog, logger.GCPFieldKeyTime)
+	require.NotEmpty(t, tw.lastLog[logger.GCPFieldKeyMsg])
+
+	log.Debug().Msg("is it on?")
+	require.Equal(t, uint16(1), tw.levels[zerolog.TraceLevel])
```

Comment:
Should we be checking the `zerolog.DebugLevel` entry here? Same for the Info, Warn, Error, Fatal, etc. below but with the respective log levels.

Reply:
Whoops, you're right - good catch!